packages/jest-haste-map/src/index.js

```diff
@@ -246,12 +248,16 @@ class HasteMap extends EventEmitter {
         : null,
       name: options.name,
       platforms: options.platforms,
+      preserveSymlinks: options.preserveSymlinks,
       resetCache: options.resetCache,
       retainAllFiles: options.retainAllFiles,
       rootDir: options.rootDir,
       roots: Array.from(new Set(options.roots)),
       throwOnModuleCollision: !!options.throwOnModuleCollision,
-      useWatchman: options.useWatchman == null ? true : options.useWatchman,
+      // Watchman can not handle symlinks: https://github.com/facebook/watchman/issues/105
+      useWatchman: options.preserveSymlinks == true
```

Comment:
We should throw a hard error somewhere if both Watchman and preserve symlink options are provided